test/Grpc.AspNetCore.Server.Tests/HttpContextServerCallContextTests.cs

```diff
@@ -55,12 +57,29 @@ public void WriteResponseHeadersAsyncCore_AddsMetadataToResponseHeaders()
 
             // Act
             var serverCallContext = new HttpContextServerCallContext(httpContext);
-            serverCallContext.WriteResponseHeadersAsync(metadata);
+            await serverCallContext.WriteResponseHeadersAsync(metadata);
 
             // Assert
             Assert.AreEqual("bar", httpContext.Response.Headers["foo"]);
         }
 
+        [Test]
+        public async Task WriteResponseHeadersAsyncCore_Base64EncodesBinaryResponseHeaders()
+        {
+            // Arrange
+            var headerBytes = new byte[] { 0x01, 0x02, 0x03, 0x04 };
+            var httpContext = new DefaultHttpContext();
+            var metadata = new Metadata();
+            metadata.Add("foo-bin", headerBytes);
```

Comment:
What happens for `FOO-BIN`?

Reply:
An error is thrown by Metadata.Add(). Since all binary headers must have `-bin` suffix with that exact casing. I don't think it's necessary to add a test for this since it's an implementation detail in Metadata/Metadata.Entry.

Reply:
Oops, I was testing something different. The casing is normalized. Will add test case.